비교하기 기능 기본 레이아웃 구현 및 색상 속성 동시 변경 기능 구현

gitgitWi opened the pull request:
# 비교하기 기능 기본 레이아웃 구현 

- 56333a7

## 구현 내용

- 비교하기 버튼 클릭에 따른 비교하기 창 토글 구현
  - 비교하기 창 레이아웃
  - 비교하기 클릭이벤트 관련 hook
- `mapbox-gl-compare` 패키지 설치

## TODO

- 비교하기 버튼 클릭 이후 mapbox 기본 Compare 함수는 작동하고 있으나, 지도가 표시되지 않음
  - debugger 통해 데이터 이동 확인
  - before-after div 변화 확인
  - map rendering 방식 확인 필요함

---

# 색상 관련 Styler 속성 동시 변경 기능 구현 

## 구현 내용

- 32cbd10
- 색상-채도-밝기 중 1개 속성 변경시 그에 따라 다른 속성도 변경

## TODO

- 채도-밝기의 경우 input 변경에 따라 바로 변경되지만,
- 컬러는 마우스 이벤트 등 사용자 이벤트가 한번더 추가되어야 변경되는 경우가 있음
- 원인은 아직 파악 못한 상태이지만, 함수가 복잡하기 때문일 것으로 추정
- 대부분 일반 사용자 입장에서 색상-채도-밝기는 하나로 묶어도 크게 다르지 않을 것으로 판단되고,
- 별도의 color picker를 만드는 것도 좋은 대안이라 생각해 현재 작업 중

--- 

## 그 외

### CSS 조정 

- 5ec3f05
- 버튼 크기 및 border 수정:  mapbox 기본 버튼 css와 통일감 주기 위함
- `NotoSansKR`로 기본 폰트 지정: 사이드바 및 검색창 폰트에 폰트 지정하여, 브라우저 및 설정 달라도 동일한 레이아웃

### [CHORE] git commit template 파일 추가 …

- e4974a2
- Ground Rule에 따른 커밋 메시지 작성 편리를 위해 템플릿 파일 추가

Commit:
[FEAT] CSS 조정

- 버튼 크기 및 border 수정:  mapbox 기본 버튼 css와 통일감 주기 위함
- `NotoSansKR`로 기본 폰트 지정: 사이드바 및 검색창 폰트에 폰트 지정하여, 브라우저 및 설정 달라도 동일한 레이아웃

Commit:
[CHORE] git commit template 파일 추가

- Ground Rule에 따른 커밋 메시지 작성 편리를 위해 템플릿 파일 추가

Commit:
[FEAT] [#77] 비교하기 기능 관련 기본 레이아웃 구현

## 구현 내용

- 비교하기 버튼 클릭에 따른 비교하기 창 토글 구현
  - 비교하기 창 레이아웃
  - 비교하기 클릭이벤트 관련 hook
- `mapbox-gl-compare` 패키지 설치

## TODO

- 비교하기 버튼 클릭 이후 mapbox 기본 Compare 함수는 작동하고 있으나, 지도가 표시되지 않음
  - debugger 통해 데이터 이동 확인
  - before-after div 변화 확인
  - map rendering 방식 확인 필요함

Commit:
[FEAT] [#106] 색상 관련 Styler 동시에 변경

- 색상-채도-밝기 중 1개 속성 변경시 그에 따라 다른 속성도 변경

- 채도-밝기의 경우 input 변경에 따라 바로 변경되지만,
- 컬러는 마우스 이벤트 등 사용자 이벤트가 한번더 추가되어야 변경되는 경우가 있음
- 원인은 아직 파악 못한 상태이지만, 함수가 복잡하기 때문일 것으로 추정
- 대부분 일반 사용자 입장에서 색상-채도-밝기는 하나로 묶어도 크게 다르지 않을 것으로 판단되고, 별도의 color picker를 만드는 것도 좋은 대안이라고 생각됨

Commit:
[FEAT] Color Picker 구현중

- 색상-채도-밝기 통합하여 한번에 관리하고, 함수도 통일 해야 함
- 브라우저 간 color picker UI 차이 최소화

gitgitWi commented:
color picker 테스트로 구현 중이며, 색상 속성 변경 함수를 아예 하나로 합쳐서 좀더 간소하게 만들 수 있을 것 같습니다

![image](https://user-images.githubusercontent.com/57997672/101296604-9e169880-3867-11eb-9628-4992b83e13be.png)

qkrdmstlr3 commented:
저도 color picker컴포넌트를 만들어보고 있습니다!

qkrdmstlr3 commented on src/components/Map/ButtonGroup/UpperButtons.tsx:
```diff
@@ -44,7 +50,14 @@ function UpperButtons({ mapRef }: UpperButtonsProps): ReactElement {
+          height="40px"
+          onClick={
+            () => comparisonButtonClickHandler()
+            // eslint-disable-next-line react/jsx-curly-newline
```
브라켓안에 함수만 넣는 방식으로 수정가능 할 것 같습니다.

qkrdmstlr3 commented on src/components/Sidebar/Sidebar.tsx:
```diff
@@ -17,6 +17,7 @@ const SidebarWrapper = styled.div`
   height: 100vh;
   z-index: 10;
   background-color: white;
+  font-family: 'Noto Sans KR', sans-serif;
```
👍

qkrdmstlr3 commented on src/hooks/sidebar/useStyleType.ts:
```diff
@@ -11,13 +11,66 @@ import {
+
+  return newStyleObj;
+};
+
```
따로 util파일로 빼는 것은 어떨까요?

Eunsol0410 commented on src/components/Map/Map.tsx:
```diff
@@ -22,14 +30,57 @@ const MapWrapper = styled.div`
+      transform: translateX(0);
+      opacity: 1;
+    }
+  }
```
이 부분 설명해주실 수 있나요?! 궁금합니다 ㅎ

Eunsol0410 commented on src/hooks/map/useComparisonButton.ts:
```diff
@@ -0,0 +1,77 @@
+): useComparisonButtonType {
+  const [isComparisonToggle, setIsComparisonToggle] = useState(false);
+  const afterMapRef = useRef<HTMLDivElement>(null);
+  const comparisonMapRef = useRef<HTMLDivElement>(null);
```
현재 스토어의 맵 객체와 해당 ref를 사용해야 하지 않을까요??? 🤔

qkrdmstlr3 commented on src/components/Map/Map.tsx:
```diff
@@ -22,14 +30,57 @@ const MapWrapper = styled.div`
+        ref={afterMapRef}
+        isComparisonToggle={isComparisonToggle}
+      />
+    </MapsWrapper>
```
compareRef의 div가 하위 div두개를 자동으로 compare하게 만들어주는 구조인가요?

qkrdmstlr3 reviewed (commented):
수고하셨습니다!
혹시 초기 Saturation / Lightness값도 색상에 따라서 변경이 되나요??

GodDrinkTeJAVA commented on src/components/Map/Map.tsx:
```diff
@@ -22,14 +30,57 @@ const MapWrapper = styled.div`
+  right: 0;
+  z-index: 1;
+
+  display: ${({ isComparisonToggle }) =>
+    isComparisonToggle ? 'absolute' : 'none'};
```
이 부분에서 isComparisonToggle이 값이 참이라면 display의 값이 absolute가 되도록 되어있는데 absolute가 들어갈 수 있는 속성은 position 아니던가요?? 🕵️

Eunsol0410 reviewed (commented):
컬러는 색상표가 닫힘에 따라 적용되어서 onStyleChange가 불려지는데, 정훈님이 PR에 언급하신 부분을 정확히 이해하지 못했습니다 😂 설명부탁드립니다!

Commit:
Merge branch 'feature/map-style' of https://github.com/boostcamp-2020/Project08-A-Styled-Map-Admin-Tool into feature/map-styling

Commit:
Merge branch 'feature/map-style' of https://github.com/boostcamp-2020/Project08-A-Styled-Map-Admin-Tool into feature/map-styling

Commit:
[FIX] ColorStyle 바로 적용안되는 에러 수정

- EventListner를 `onBlur` -> `onMouseLeave`로 수정하여 임시로 해결
- OS/Browser 문제인지는 잘 모르겠습니다
  - 현재 개발 환경: Ubuntu20.04, Chrome 및 Opera

gitgitWi commented:
- d10002d 에서 ColorStyle 바로 적용 안되는 에러 임시로 수정했습니다.

gitgitWi committed:
Merge branch 'feature/map-style' into feature/map-styling
